crates/wysiwyg/src/composer_model/example_format.rs

```diff
@@ -368,41 +368,58 @@ impl SelectionWritingState {
         self.current_pos += code_units;
 
         // If we just passed first, write out {
-        let do_first = !self.done_first && self.first < self.current_pos;
+        let mut do_first = !self.done_first && self.first < self.current_pos;
 
         // If we just passed last or we're at the end, write out }
         let do_last = !self.done_last
             && (self.last <= self.current_pos
                 || self.current_pos == self.length);
 
+        // In some weird circumstances with empty text nodes, we might
+        // do_last when we haven't done_first, so make sure we do_first too.
+        if do_last && !self.done_first {
```

Comment:
Should we add a TODO to delete this when we properly handle empty text nodes?

Reply:
I'm not 100% sure we will want to delete it, and this is pretty much test code, so I'll avoid adding yet more TODOs in this case, I think.